.travis.yml

```diff
@@ -1,12 +1,19 @@
+dist: xenial
```

Comment:
Had to do some Travis work since the build seems to have rotted

* Upgrade to a newer Ubuntu version with more modern JDK's installed
* Change openjdk7 and oraclejdk8 to openjdk8 since the Travis install JDK script now fails on JDKs <9 (openjdk8 is preinstalled)
* Downloading SBT takes very long and frequently times out (see builds of now-replaced commits), caching the `~/.sbt` directory should work around this (once it succeeds once it can reuse the cache) and is a good idea anyway since it increases the build speed. `~/.ivy2/cache` is there to cache dependencies for build speed.